### PR TITLE
toxiproxy 2.6.0 (new formula)

### PR DESCRIPTION
**Pull request by bayandin**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by bayandin** [comment]
```
==> brew audit toxiproxy --online --new-formula
==> FAILED
Full audit toxiproxy --online --new-formula output
  Error: GitHub API Error: Although you appear to have the correct authorization credentials, the `Shopify` organization has an IP allow list enabled, and your IP address is not permitted to access this resource.
  HOMEBREW_GITHUB_API_TOKEN may be invalid or expired; check:
    https://github.com/settings/tokens
```

The same audit failure as in https://github.com/Homebrew/homebrew-core/pull/137785#issuecomment-1653199167

It looks like all Shopify formulae (2) are affected

```
$ git grep 'shopify'  
Formula/s/shadowenv.rb:  homepage "https://shopify.github.io/shadowenv/"
Formula/t/toxiproxy.rb:  homepage "https://github.com/shopify/toxiproxy"
```

**Review by chenrui333 — APPROVED** [review]

**Review by ZhongRuoyu — COMMENTED** [review]
Test failed on Linux. Probably need

```ruby
uses_from_macos "ruby" => :test
```

File: Formula/t/toxiproxy.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
+class Toxiproxy < Formula
+  desc "TCP proxy to simulate network & system conditions for chaos & resiliency testing"
+  homepage "https://github.com/shopify/toxiproxy"
+  url "https://github.com/Shopify/toxiproxy/archive/refs/tags/v2.6.0.tar.gz"
+  sha256 "17bf7580644a5cf8d6fc2b5a71e8dc7931528838724a6e4dc2bb326731fab4c7"
+  license "MIT"
+
+  depends_on "go" => :build
+  uses_from_macos "ruby" => :build
```
Comment (SMillerDev) [inline Formula/t/toxiproxy.rb:9]:
```suggestion
  uses_from_macos "ruby" => :test
```

Comment (bayandin) [reply Formula/t/toxiproxy.rb]:
```
==> ruby /tmp/toxiproxy-test-20230824-9657-128oi22/hello-homebrew.rb
<internal:/home/linuxbrew/.linuxbrew/Cellar/ruby/3.2.2_1/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:85:in `require': cannot load such file -- webrick (LoadError)
	from <internal:/home/linuxbrew/.linuxbrew/Cellar/ruby/3.2.2_1/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
	from /tmp/toxiproxy-test-20230824-9657-128oi22/hello-homebrew.rb:1:in `<main>'
```

Let me just start it from the test in a separate thread.

**Pull request merged**